src/utilities/get-program-path-by-site-code.js

```diff
@@ -0,0 +1,27 @@
+import { useStaticQuery, graphql } from "gatsby"
+
+export const useGetProgramPathBySiteCode = () => {
+  const allCentralProgramNodes = useStaticQuery(
+    graphql`
+      query {
+        allCentralProgramsJson {
+          nodes {
+            code
+            fields {
+              path
+            }
+          }
+        }
+      }
+    `
+  )
+  let pathsBySiteCode = {}
+
+  allCentralProgramNodes.allCentralProgramsJson.nodes.forEach(program => {
+    pathsBySiteCode[program.code] = program.fields.path
+  })
```

Comment:
I can't tell if this gets run once or every time it's used. If every time, I guess we could just use `find` in the call below?

Reply:
Oh I see. I didn't know about `useMemo`, thanks. Probably not an optimization we need? But I have a side goal to learn how to do things "The Right Way" in React. 😛 Do you think it would hurt?

Cool, yeah. Could just return the path. Is there a convention that you need the `use` prefix in the domain? Or could the function name just become `getProgramPathBySiteCode`?

Reply:
I think useMemo is best used when you have something that's really computationally heavy or when you need to have an object reference persist between renders (e.g. for dependency lists for hooks). I've seen a couple [blog posts](https://kentcdodds.com/blog/usememo-and-usecallback) saying that often times the change in performance isn't worth complicating the code.

I'm pretty sure you have to have the use prefix when defining a custom hook, it might even throw an error if you don't.

I also wanted to ask: what do you think about the destructuring assignment ```const {allCentralProgramsJson: {nodes}} = useStaticQuery```? Do you feel like it hurts readability?

Reply:
Tried to change the name and answered my own question. Looks like it needs to start with `use`.

Reply:
oops, sorry - I missed your earlier comment. Put in a useMemo, but I can remove it. Still need to do the Contenful part where this component might actually be used.

I did a double take when I saw the deconstructed assignment, but I think I get it now. I actually haven't seen it before. I rarely write javascript (or any code) these days. 😅 But it's pretty slick.